src/Tizen.NUI/src/public/GlWindowEvent.cs

```diff
@@ -0,0 +1,454 @@
+                    return _touch;
+                }
+                set
+                {
+                    _touch = value;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Key event arguments.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public class KeyEventArgs : EventArgs
+        {
+            private Key _key;
+
+            /// <summary>
+            /// Key.
+            /// </summary>
+            [EditorBrowsable(EditorBrowsableState.Never)]
+            public Key Key
+            {
+                get
+                {
+                    return _key;
+                }
+                set
+                {
+                    _key = value;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Feeds a key event into the window.
+        /// This resized event arguments.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public class ResizedEventArgs : EventArgs
+        {
+            Size2D _windowSize;
+
+            /// <summary>
+            /// This window size.
+            /// </summary>
+            [EditorBrowsable(EditorBrowsableState.Never)]
+            public Size2D WindowSize
+            {
+                get
+                {
+                    return _windowSize;
+                }
+                set
+                {
+                    _windowSize = value;
+                }
+            }
+        }
+
+        /// <summary>
+        /// VisibilityChangedArgs
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public class VisibilityChangedArgs : EventArgs
+        {
+            private bool visibility;
+            /// <summary>
+            /// Visibility
+            /// </summary>
+            [EditorBrowsable(EditorBrowsableState.Never)]
+            public bool Visibility
+            {
+                get => visibility;
+                set {
+                    visibility = value;
+                }
+            }
+        }
+
+        private void OnVisibilityChanged(IntPtr window, bool visibility)
+        {
+            if (window == global::System.IntPtr.Zero)
+            {
+                NUILog.Error("[ERR] OnVisibilityChanged() window is null");
+                return;
+            }
+
+            VisibilityChangedArgs e = new VisibilityChangedArgs();
+            e.Visibility = visibility;
+            if (VisibilityChangedEventHandler != null)
+            {
+                VisibilityChangedEventHandler.Invoke(this, e);
+            }
+        }
+
+        [UnmanagedFunctionPointer(CallingConvention.StdCall)]
+        private delegate void GlVisibilityChangedEventCallbackType(IntPtr window, bool visibility);
+        private GlVisibilityChangedEventCallbackType GlVisibilityChangedEventCallback;
+        private event EventHandler<VisibilityChangedArgs> VisibilityChangedEventHandler;
```

Comment:
this is trivial one. but could you decide whether underscore( _ ) is used for private variable or not?
some has underscore but this one doesn't have it... ^^;